### PR TITLE
Add Remove and Add Functionality

### PR DESCRIPTION
**Pull request by Munsa1**
###Remove and Add Functionality

In the following milestone,  I :

- Remove all hardcoded items from the tasks array.

- Create a new JavaScript file for the new functionality.

- Implement a function for adding a new task (add a new element to the array).

- Implement a function for editing task descriptions.

- Implement a function for deleting a task (remove an element from the array).

- Implement a function for the "Clear all completed" button (use filter() method).

Take a look👍

### CODE REVIEW

**Review by henatan99 — CHANGES_REQUESTED** [review]
Hi @Munsa1 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
- [ ] While editing a task description, hitting enter will make a new line and the local storage adds an enter character to the edited task description. Kindly fix this. :pray: 

 
Screenshot from 2021-10-29 23-23-27

_Check the comments under the review._

## Optional suggestions
- [ ] OPTIONAL: Your PR has no changed files. It means that you made no change to this milestone at all. As far as I understand it is because you have done this milestone/branch requirements in the previous milestone/branch. I recommend you do not do requirements ahead of the current milestone you are working on and always carefully read the project milestone requirements. 

image

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/add_remove.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import { fromStorage, toStorage, reloadStore } from './store';
+// eslint-disable-next-line import/no-cycle
+import { dragHover } from './dragdrop';
+import { refreshStore } from './status';
+
+const editHandlers = () => {
+  const todoList = document.getElementsByClassName('todo-item');
+  for (let i = 0; i < todoList.length; i += 1) {
+    const labelElem = todoList[i].children[0].children[1];
+    labelElem.addEventListener('input', () => {
```
Comment (bellom) [inline src/add_remove.js:17]: Hi, This is an additional review as a result of the second opinion request.

The CR meant when the user is trying to edit a task, the keypress `Enter` is saved in the localStorage, and also created a new line on the UI.

To reproduce this error: try to edit one of the tasks, and press `Enter` key.

Kindly check the screenshot from your app below;
image


**_Hint_**: Try to check when `key = 'Enter'` 

`if (e.key === 'Enter') {...}`

**Comment by bellom** [comment]
Hello 👋,

This is an additional review as a result of the second opinion request.

Let's try to use inline comments to explain, sorry if the previous CR wasn't clear enough.🙏 

@Henok let's try to use the GitHub inline comment feature to explain changes, that way, you're pointing to where the problem is coming from.

Check my comment above the review.

Kindly send in a re-review once you're done!

Keep rocking 🚀!

**Comment by henatan99** [comment]
> Hello wave,
> 
> This is an additional review as a result of the second opinion request.
> 
> Let's try to use inline comments to explain, sorry if the previous CR wasn't clear enough.pray
> 
> @henok let's try to use the GitHub inline comment feature to explain changes, that way, you're pointing to where the problem is coming from.
> 
> Check my comment above the review.
> 
> Kindly send in a re-review once you're done!
> 
> Keep rocking rocket!

@bellom , thanks for the recommendation. In fact, I always use the Github review comment feature. But if you look carefully at the review summary I made, this PR did not have any changed file, as a result, I could not make an inline review comment. I thought he already made this milestone requirements in the previous milestone and he submitted the CRR without making any change from the previous milestone, and just reviewed it. :pray:

File: src/add_remove.js (hunk 1 above)
Comment (henatan99) [reply src/add_remove.js:17]: This issue is well addressed. Great job! :clap: :clap:

**Review by henatan99 — APPROVED** [review]
Hi @Munsa1 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**